### PR TITLE
support nacos metadata

### PR DESCRIPTION
**Pull request by Moriadry**
## What is the purpose of the change

complete nacos metadata report implement.

## Brief changelog

nacos metadata report impl, and code clean (switch to using a nacos utils class)

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/incubator-dubbo-samples) project.
- [x] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Moriadry** [comment]
@ralf0131  could pls help me check this? I am wondering whether i miss something. Thanks!

File: dubbo-metadata-report/dubbo-metadata-report-nacos/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.metadata.store.MetadataReportFactory [hunk 1]
```diff
@@ -0,0 +1 @@
+consul=org.apache.dubbo.metadata.store.consul.ConsulMetadataReportFactory
```
Comment (ralf0131) [inline dubbo-metadata-report/dubbo-metadata-report-nacos/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.metadata.store.MetadataReportFactory]: Should be change to nacos

File: dubbo-common/pom.xml [hunk 2]
```diff
@@ -77,5 +77,9 @@
             <groupId>de.ruedigermoeller</groupId>
             <artifactId>fst</artifactId>
         </dependency>
+        <dependency>
```
Comment (ralf0131) [inline dubbo-common/pom.xml]: dubbo-common is the most basic module that all the other modules might depend on, I think it should be depend on nacos-client here. It looks weird.

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/NacosUtils.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import static com.alibaba.nacos.api.PropertyKeyConst.NAMESPACE;
+import static com.alibaba.nacos.api.PropertyKeyConst.CLUSTER_NAME;
+import static com.alibaba.nacos.client.naming.utils.UtilAndComs.NACOS_NAMING_LOG_NAME;
+
+import static org.apache.dubbo.common.Constants.BACKUP_KEY;
+
+/**
+ * NacosUtils
+ */
+public class NacosUtils {
```
Comment (ralf0131) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/NacosUtils.java]: I could not find a better location to put this class. Maybe we should put this class into nacos-client. Before that I think it is ok to keep it duplicated. How do you think?

Comment (Moriadry) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/NacosUtils.java]: yeah, your right. Let me fix this

**Comment by ralf0131** [comment]
Hi, I am starting to look at this.

**Comment by ralf0131** [comment]
You need to add `<module>dubbo-metadata-report-nacos</module>` to `dubbo-metadata-report/pom.xml`...

**Comment by ralf0131** [comment]
Hi, your code has compile errors, please fix it..

**Comment by Moriadry** [comment]
@ralf0131 yes, push it. Sorry for missing this.

**Comment by ralf0131** [comment]
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18:51 min
[INFO] Finished at: 2019-05-13T09:56:44Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project dubbo-metadata-report-nacos: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/apache/incubator-dubbo/dubbo-metadata-report/dubbo-metadata-report-nacos/src/test/java/org/apache/dubbo/metadata/store/nacos/NacosMetadataReportTest.java:[47,39] error: cannot find symbol
[ERROR]   symbol:   variable SESSION_TIMEOUT_KEY
[ERROR]   location: class Constants
[ERROR] /home/travis/build/apache/incubator-dubbo/dubbo-metadata-report/dubbo-metadata-report-nacos/src/test/java/org/apache/dubbo/metadata/store/nacos/NacosMetadataReportTest.java:[92,79] error: cannot find symbol
[ERROR]   symbol:   variable PROVIDER_SIDE
[ERROR]   location: class Constants
[ERROR] /home/travis/build/apache/incubator-dubbo/dubbo-metadata-report/dubbo-metadata-report-nacos/src/test/java/org/apache/dubbo/metadata/store/nacos/NacosMetadataReportTest.java:[104,111] error: cannot find symbol
[ERROR]   symbol:   variable CONSUMER_SIDE
[ERROR]   location: class Constants
[ERROR] -> [Help 1]
```

**Comment by Moriadry** [comment]
@ralf0131 hi, I merged with remote/master, can you check again.

**Comment by ralf0131** [comment]
This pull request is targeted to solve #3846

**Pull request merged**